git_wrapper/log.py

```diff
@@ -107,3 +127,26 @@ def grep_for_commits(self, branch, grep_for, reverse=False, path=None):
             commits = results.split("\n")
 
         return commits
+
+    @reference_exists('commit_ref')
+    def log_show_commit(self, commit_ref='HEAD', pattern="$full_message"):
+        """Return a string representing the given commit.
+
+           Any of the following placeholders may be used in the pattern:
+             * $hash The full commit hash
+             * $short_hash The short commit hash, similar to --abbrev-commit
+             * $message The commit message
+             * $summary First line of the commit message
+             * $full_message Complete commit info with hash, author, message.
+               Similar to default "git log" ouput
+             * $author Commit author
+             * $date Date the commit was authored
+
+           :param str commit_ref: A commit hash. Defaults to HEAD
+           :param str pattern: Formatter containing any of the placeholders above
+           :return: A string
+        """
+        result = self.git_repo.repo.commit(commit_ref)
+        if result:
+            return self.commit_format([result], pattern)[0]
+        return ''
```

Comment:
Can we ever hit that, though? No results means the commit_ref was wrong, but that's caught by the decorator now?

Reply:
well... I did hit this in the unit tests (tox -e py27) actually..... maybe I did it wrong though? If you see why "test_log_show_commit_no_commit" didn't fail with the Exception....